frame/asset-conversion/src/lib.rs

```diff
@@ -736,18 +718,19 @@ pub mod pallet {
 		}
 
 		pub(crate) fn do_swap(
-			sender: &T::AccountId,
+			sender: T::AccountId,
 			amounts: &Vec<T::AssetBalance>,
-			path: &BoundedVec<T::MultiAssetId, T::MaxSwapPathLength>,
-			send_to: &T::AccountId,
+			path: BoundedVec<T::MultiAssetId, T::MaxSwapPathLength>,
+			send_to: T::AccountId,
 			keep_alive: bool,
 		) -> Result<(), DispatchError> {
-			if let Some([asset1, asset2]) = path.get(0..2) {
+			ensure!(amounts.len() > 1, Error::<T>::InvalidPath);
+			if let Some([asset1, asset2]) = &path.get(0..2) {
```

Comment:
Instead of using if .. else we can simply write here:
`let [asset1, asset2] = &path.get(0..2).ok_or(Error::<T>::CorrespondenceError)?;` (or Error::<T>:: InvalidPath)

Reply:
Unfortunately that's not an irrefutable pattern. Using a match is either more verbose or makes it one more indentation more. I think this is as good as it gets for now.